### ISSUE TITLE
Randomizer rules for the 3.0 Update

### ISSUE THREAD

**Issue by Remerod** [jj248/RealmsInExile#782]
So I think we need some updates for our randomizer functions/rules for 3.0 (as I've been speaking with others about) 

So while the rules still work currently some adaptations might be advised to take for them

* Make Valinor an exception of the rule. - the Valinor courtiers/rulers should not be part of **Randomized Faiths** or **Random Ruler placement**, or even **Shattered Earth** rule. Since they serve as a non-playable area and are more there for lore stuff (Elrond and similar should probably also be taken into account for this).

* Make so random faith for Elves only switches there faith inside of the elven faith group. - there is no real good reasons for elves to become cult of the lidless eye or any of the harradrim faiths, honestly it might cause bugs, so making them them stay within their faith group might be advisable.
(if we do this for elves then maybe we could do something similar for orcs and nazguls and so on)

* Elven rulers that are moved because of the **Random Ruler placement** rule will becomes Feudal government, this is most liekly a problem? i guess a solution could be to only allow them to be moved within the elven homelands but i dunno. Something should probably be done

* Special rulers removal. This is something for "if" we have the time for it, but people like Saruman and Sauron will become very strong in a **Shattered Earth** rule game, having a rule that replaces them could be very fitting, like Saruman would be replaced with an uruk-hai or something.

**Comment by jj248** [comment]
OK - thinking about this:

- Agreed on the first part.  We should make sure Valinor is untouched by these game rules.
- I'm not inclined to restrict religion randomness.  If you're using the game rule, I think you're up for 'anything goes' rather than a lore-friendly run.
- Managing changing of holding types will be a nightmare under random ruler placement, and even more so when we add more races, so I think I'll just remove this game option, especially given I'm sure it's very very rarely used.
- I can see the argument for removing special rulers, but personally, I really like them.  There is zero challenge to a Shattered Earth run without them, as the player is just so much better at expansion and consolidation than the AI, so I'm inclined just to leave them in for now. :)

So in summary, I'll:

- [ ] Exempt Valinor from random faiths and shattering in shattered earth rulesets.
- [ ] Remove Random Ruler placement rule.

**Comment by jj248** [comment]
Implemented - ready to test.

**Comment by Remerod** [comment]
>Managing changing of holding types will be a nightmare under random ruler placement, and even more so when we add more races, so I think I'll just remove this game option, especially given I'm sure it's very very rarely used.

wouldn't it just be better to leave it as it is then, just a bit broken, rather then completely remove it, it is rarelt used yeah but it is fun sometimes

**Comment by Remerod** [comment]
> Exempt Valinor from random faiths and shattering in shattered earth rulesets.

seems to be working fine 
![image](https://user-images.githubusercontent.com/79544635/234732857-37110901-d508-41e8-97a5-daf8722d8e7f.png)

> Remove Random Ruler placement rule.

Seems to be working but i retract to my previous comment

**Comment by Remerod** [comment]
closing for now, will return for the Remove Random Ruler placement rule.
